src/util/merkleblock.rs

```diff
@@ -420,32 +421,40 @@ impl MerkleBlock {
     /// // Create a merkle block containing a single transaction
     /// let txid = Txid::from_hex(
     ///     "5a4ebf66822b0b2d56bd9dc64ece0bc38ee7844a23ff1d7320a88c5fdb2ad3e2").unwrap();
-    /// let match_txids = vec![txid].into_iter().collect();
-    /// let mb = MerkleBlock::from_block(&block, &match_txids);
+    /// let match_txids: Vec<Txid> = vec![txid].into_iter().collect();
+    /// let mb = MerkleBlock::from_block_with_predicate(&block, |t| match_txids.contains(t));
     ///
     /// // Authenticate and extract matched transaction ids
     /// let mut matches: Vec<Txid> = vec![];
     /// let mut index: Vec<u32> = vec![];
     /// assert!(mb.extract_matches(&mut matches, &mut index).is_ok());
     /// assert_eq!(txid, matches[0]);
     /// ```
-    pub fn from_block(block: &Block, match_txids: &HashSet<Txid>) -> Self {
+    pub fn from_block_with_predicate<F>(block: &Block, match_txids: F) -> Self
+        where F: Fn(&Txid) -> bool {
         let block_txids: Vec<_> = block.txdata.iter().map(Transaction::txid).collect();
-        Self::from_header_txids(&block.header, &block_txids, match_txids)
+        Self::from_header_txids_with_predicate(&block.header, &block_txids, match_txids)
     }
 
-    /// Create a MerkleBlock from the block's header and txids, that should contain proofs for match_txids.
+    /// Create a MerkleBlock from a block, that contains proofs for specific txids.
+    #[cfg(feature = "std")]
+    #[deprecated(since="0.26.2", note="use from_block_p")]
+    pub fn from_block(block: &Block, match_txids: &::std::collections::HashSet<Txid>) -> Self {
+        Self::from_block_with_predicate(block, |t| match_txids.contains(t))
+    }
+
+    /// Create a MerkleBlock from the block's header and txids, that contain proofs for specific txids.
     ///
     /// The `header` is the block header, `block_txids` is the full list of txids included in the block and
-    /// `match_txids` is a set containing the transaction ids that should be included in the partial merkle tree.
-    pub fn from_header_txids(
+    /// `match_txids` is a function that returns true for the ids that should be included in the partial merkle tree.
+    pub fn from_header_txids_with_predicate<F>(
         header: &BlockHeader,
         block_txids: &[Txid],
-        match_txids: &HashSet<Txid>,
-    ) -> Self {
+        match_txids: F,
+    ) -> Self where F: Fn(&Txid) -> bool {
         let matches: Vec<bool> = block_txids
             .iter()
-            .map(|txid| match_txids.contains(txid))
+            .map(|txid| match_txids(txid))
```

Comment:
Minor nit: wondering if you could directly pass the `match_txids` instead of calling it from a closure.

Reply:
:+1: